modules/bluefish/util/blue_velvet.cpp

```diff
@@ -23,224 +23,266 @@
 #include "blue_velvet.h"
 #include <common/utf.h>	
 #include <core/video_format.h>
-#include <bluefish/interop/BlueVelvetCUtils.h>
+#include <BlueVelvetCUtils.h>
 
 #if defined(__APPLE__)
-	#include <dlfcn.h>
+#include <dlfcn.h>
 #endif
 
 #if defined(_WIN32)
-	#define GET_PROCADDR_FOR_FUNC(name, module) { name = (pFunc_##name)GetProcAddress(module, #name); if(!name) { return false; } }
+	#define GET_PROCADDR_FOR_FUNC(name, module) { name = (pFunc_##name)GetProcAddress(reinterpret_cast<HMODULE>(module), #name); if(!name) { return false; } }
+//	#define GET_PROCADDR_FOR_FUNC(name, module) { name = (pFunc_##name)GetProcAddress(module, #name); if(!name) { return false; } }
 #elif defined(__APPLE__)
 	#define GET_PROCADDR_FOR_FUNC(name, module) { name = (pFunc_##name)dlsym(module, #name); if(!name) { return false; } }
 #endif
 
 namespace caspar { namespace bluefish {
 
-	BvcWrapper::BvcWrapper()
+	bvc_wrapper::bvc_wrapper()
 	{
-		bvc = nullptr;
-		hModule = nullptr;
-		InitFunctionsPointers();
-		if (bfcFactory)
-			bvc = bfcFactory();
+		bvc_ = nullptr;
+		h_module_ = nullptr;
+		init_function_pointers();
+		bvc_ = std::shared_ptr<void>(bfcFactory(), bfcDestroy);
```

Comment:
Access violation occurs here because if `init_function_pointers()` returns `false` `bfcFactory` will be `nullptr`. You should instead throw the exception `if (!init_function_pointers())`. But also if `bfcFactory()` returns `nullptr`.